### PR TITLE
Improv/better enums

### PR DESCRIPTION
**Pull request by lbonn**
Big but (hopefully) dumb.

Also, I've switched back to define a separate variable for the current sql schema version. I think I suggested removing it because of the duplication but in fact, it simplifies things a bit.

### CODE REVIEW

**Comment by patrickvacek** [comment]
While personally I'm not particularly invested, for consistency, we should indeed follow the Google standard:

https://google.github.io/styleguide/cppguide.html#Enumerator_Names

So I think if we're going in and modernizing all of the variations we have, we might as well make them all consistent: `EnumClassName::kValueName`.

But otherwise, looks good!

File: src/sota_tools/authenticate.cc [hunk 1]
```diff
@@ -8,17 +8,17 @@ using std::string;
 
 int authenticate(const string &cacerts, const ServerCredentials &creds, TreehubServer &treehub) {
   switch (creds.GetMethod()) {
-    case AUTH_BASIC: {
+    case AuthMethod::BASIC: {
```
Comment (patrickvacek) [inline src/sota_tools/authenticate.cc]: Sorry to keep bugging, but can we rename these all-caps ones, too?

Comment (lbonn) [reply src/sota_tools/authenticate.cc]: So, `AuthMethod::kBasic`?

Comment (patrickvacek) [reply src/sota_tools/authenticate.cc]: Yeah, I think that's best.

**Pull request merged**